tests/cpp/cell/test_gemm.cu

```diff
@@ -45,150 +203,124 @@ __global__ void test_wmma1(const Element* ga, const Element* gb, ElementAcc* gc,
     RegC acc;
 
     for (int k = 0; k < TileIteratorA::sc1; ++k) {
-        load_rA(sAs(_, k).to_tile(), rA);
-        // Since we interpret a column-major B with a shape of [K, N] as a
-        // row-major B with a shape of [N, K], here sBs is sliced along the
-        // first dimension instead of the second one.
-        load_rB(sBs(_, k).to_tile(), rB);
+        load_rA(sAs(k), rA);
```

Comment:
Is this mean automatic tiling based on row-major/column-major?

Reply:
So once `sAs` is defined, it actually specifies which axis to start tiling along, and k indicates taking the kth index to obtain a subtile.

Reply:
> So once `sAs` is defined, it actually specifies which axis to start tiling along, and k indicates taking the kth index to obtain a subtile.

Yes, that's correct. For example, in the current implementation, the shared memory tile for operand A is defined as a $[M, K]$ row-major tile. It is then chunked into $[M, \text{stride_k}]$ segments by defining a `TileIterator`.

```cpp
using SharedA = SharedTile<Element, tl::RowMajor<M, K>>;
using TileIteratorA = SharedTileIterator<SharedA, TileShape<M, stride_k>>;
```

We then obtain a 1D list of sub-tiles, each with a shape of $M \times \text{stride_k}$. We can iterate over these tiles using a for-loop.

Reply:
An Iterator is created by specifying how to partition each dimension since Tiles are only 2D. After defining the partition, one can then use a logical index to access the sub-tiles.

Using the previous example:

```cpp
using SharedA = SharedTile<Element, tl::RowMajor<M, K>>;
using TileIteratorA = SharedTileIterator<SharedA, TileShape<stride_m, stride_k>>;

for (int k = 0; k < Iterator::sc1; ++k)
    sAs(_, k) // This returns an Iterator. To participate in computations, you need to call the `to_tile` method.
    sAs(_, k).to_tile()
```

Reply:
Got it.